pyccel/ast/numpy_wrapper.py

```diff
@@ -0,0 +1,289 @@
+                            body      = [],
+                            arguments = [Variable(dtype=PyccelPyArrayObject(), name = 'o', is_pointer=True)],
+                            results   = [Variable(dtype=NativeInteger(), name = 'i', precision = 4)])
+
+PyArray_CheckScalar   = FunctionDef(name      = 'PyArray_CheckScalar',
+                                    body      = [],
+                                    arguments = [Variable(dtype=PyccelPyObject(), name = 'o', is_pointer=True)],
+                                    results   = [Variable(dtype=NativeBool(), name = 'r')])
+
+PyArray_ScalarAsCtype = FunctionDef(name      = 'PyArray_ScalarAsCtype',
+                                    body      = [],
+                                    arguments = [Variable(dtype=PyccelPyObject(), name = 'o', is_pointer=True),
+                                                Variable(dtype=NativeVoid(), name = 'c', is_pointer = True)],
+                                    results   = [])
+
+
+numpy_flag_own_data     = Variable(dtype=NativeInteger(),  name = 'NPY_ARRAY_OWNDATA')
+numpy_flag_c_contig     = Variable(dtype=NativeInteger(),  name = 'NPY_ARRAY_C_CONTIGUOUS')
+numpy_flag_f_contig     = Variable(dtype=NativeInteger(),  name = 'NPY_ARRAY_F_CONTIGUOUS')
+
+# https://numpy.org/doc/stable/reference/c-api/dtype.html
+numpy_bool_type         = Variable(dtype=NativeInteger(),  name = 'NPY_BOOL', precision = 4)
+numpy_byte_type         = Variable(dtype=NativeInteger(),  name = 'NPY_BYTE', precision = 4)
+numpy_ubyte_type        = Variable(dtype=NativeInteger(),  name = 'NPY_UBYTE', precision = 4)
+numpy_short_type        = Variable(dtype=NativeInteger(),  name = 'NPY_SHORT', precision = 4)
+numpy_ushort_type       = Variable(dtype=NativeInteger(),  name = 'NPY_USHORT', precision = 4)
+numpy_int_type          = Variable(dtype=NativeInteger(),  name = 'NPY_INT32', precision = 4)
+numpy_uint_type         = Variable(dtype=NativeInteger(),  name = 'NPY_UINT', precision = 4)
+numpy_long_type         = Variable(dtype=NativeInteger(),  name = 'NPY_LONG', precision = 4)
+numpy_ulong_type        = Variable(dtype=NativeInteger(),  name = 'NPY_ULONG', precision = 4)
+numpy_longlong_type     = Variable(dtype=NativeInteger(),  name = 'NPY_INT64', precision = 4)
+numpy_ulonglong_type    = Variable(dtype=NativeInteger(),  name = 'NPY_ULONGLONG', precision = 4)
+numpy_float_type        = Variable(dtype=NativeInteger(),  name = 'NPY_FLOAT', precision = 4)
+numpy_double_type       = Variable(dtype=NativeInteger(),  name = 'NPY_DOUBLE', precision = 4)
+numpy_longdouble_type   = Variable(dtype=NativeInteger(),  name = 'NPY_LONGDOUBLE', precision = 4)
+numpy_cfloat_type       = Variable(dtype=NativeInteger(),  name = 'NPY_CFLOAT', precision = 4)
+numpy_cdouble_type      = Variable(dtype=NativeInteger(),  name = 'NPY_CDOUBLE', precision = 4)
+numpy_clongdouble_type  = Variable(dtype=NativeInteger(),  name = 'NPY_CLONGDOUBLE', precision = 4)
```

Comment:
why all those objects have the same precision ?, and it is good for compatibility ?.

Reply:
These objects all have the same type. They do not represent a bool object or a float object etc, but are just flags that numpy uses to tell us which datatype is saved in the array. (This code is not new, it is just moved from ast/cwrapper.py to split the python and the numpy things)

The compatibility question is a good one though. Are these objects stored in an `int`? According to [this](https://numpy.org/doc/1.17/reference/c-api.dtype.html) they are enumerate values, so `int` makes sense to me